src/env.h

```diff
@@ -601,10 +601,10 @@ class Environment : public MemoryRetainer {
   template <typename T>
   static inline T* GetBindingData(v8::Local<v8::Context> context);
 
-  typedef std::unordered_map<
-      FastStringKey,
-      BaseObjectPtr<BaseObject>,
-      FastStringKey::Hash> BindingDataStore;
+  typedef std::array<BaseObjectPtr<BaseObject>,
+                     static_cast<size_t>(
+                         EmbedderObjectType::kEmbedderObjectTypeCount)>
```

Comment:
My understanding is that `EmbedderObjectType` would be expected to expand to include non-BindingData objects eventually – should this maybe be using a separate list?

Reply:
Oh, yeah it actually already does (for `util::WeakReference`). I added a new list and a new field (`binding_type_int`) instead, with a assertion to make sure that `binding_type_int` and `type_int` match.